commons/impl/src/java/org/sakaiproject/commons/impl/PersistenceManagerImpl.java

```diff
@@ -247,6 +258,73 @@ public void run() {
         return sqlService.transact(transaction, "COMMONS_POST_DELETION_TRANSACTION");
     }
 
+    public boolean likePost(String postId, boolean toggle, String userId){
+        PostLike likeNow = getLike(postId, userId);
+        if(likeNow == null){    //if there is no existing Like, write a new one with the information provided
+            sqlService.dbWrite(POST_LIKE,new Object[]{userId, postId, toggle, new Timestamp(new Date().getTime())});
+            return true;
+        }
+        sqlService.dbWrite(LIKE_UPDATE, new Object[]{!likeNow.isLiked(), new Timestamp(new Date().getTime()), userId, postId});
+        return true;
+    }
+
+    public PostLike getLike(String postId, String userId){
+        List<PostLike> results = sqlService.dbRead(LIKE_GET, new Object[]{postId, userId}, new SqlReader<PostLike>(){
+                public PostLike readSqlResultRecord(ResultSet result) {
+                    return loadPostLikeFromResult(result);
+                }
+        });
+        if(results == null || results.size() == 0){
+            return null;
+        }
+        return results.get(0);
+    }
+
+    public int countPostLikes(String postId){
+        int likes =  sqlService.dbRead(LIKE_COUNT, new Object[]{postId}, new SqlReader<Integer>(){
+            public Integer readSqlResultRecord(ResultSet result) {
+                try{
+                    return result.getInt("COUNT(POST_ID)");
+                } catch (SQLException s){
+                    return null;
+                }
+            }
+        }).get(0);
+        return likes;
+    }
+
+    public int doesUserLike(String postId, String userId){
+        PostLike likeRecord = getLike(postId, userId);
+        if(likeRecord.isLiked()){
+            return 1;
+        }
+        return 0;
+    }
+
+    public List<PostLike> getAllUserLikes(String userId){
+        List<PostLike> results = sqlService.dbRead(LIKE_USER, new Object[]{userId}, new SqlReader<PostLike>(){
+            public PostLike readSqlResultRecord(ResultSet result) {
+                return loadPostLikeFromResult(result);
+            }
+        });
+        if(results == null){
+            return null;
+        }
+        return results;
```

Comment:
If you're returning null anyways, this if statement is unnecessary, just return results:

```suggestion
        return results;
```

Reply:
Good idea! Simplified it this way on the latest commit.